src/mrpro/operators/FiniteDifferenceOp.py

```diff
@@ -0,0 +1,139 @@
+"""Class for Finite Difference Operator."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Sequence
+from typing import Literal
+
+import torch
+
+from mrpro.operators.LinearOperator import LinearOperator
+from mrpro.utils.filters import filter_separable
+
+
+class FiniteDifferenceOp(LinearOperator):
+    """Finite Difference Operator.
+
+    This work is inspired by:
+    https://github.com/koflera/LearningRegularizationParameterMaps/blob/main/networks/grad_ops.py
+
+    Please see https://github.com/koflera/LearningRegularizationParameterMaps/tree/main?tab=Apache-2.0-1-ov-file#readme
+    for the copyright statement.
+
+    """
+
+    @staticmethod
+    def finite_difference_kernel(mode: str) -> torch.Tensor:
+        """Finite difference kernel.
+
+        Parameters
+        ----------
+        mode
+            String specifying kernel type
+
+        Returns
+        -------
+            Finite difference kernel
+
+        Raises
+        ------
+        ValueError
+            If mode is not central, forward, backward or doublecentral
+        """
+        if mode == 'central':
+            kernel = torch.tensor((-1, 0, 1)) / 2
+        elif mode == 'forward':
+            kernel = torch.tensor((0, -1, 1))
+        elif mode == 'backward':
+            kernel = torch.tensor((-1, 1, 0))
+        else:
+            raise ValueError(f'mode should be one of (central, forward, backward), not {mode}')
+        return kernel
+
+    def __init__(
+        self,
+        dim: Sequence[int],
+        mode: Literal['central', 'forward', 'backward'] = 'central',
+        pad_mode: Literal['zeros', 'circular'] = 'zeros',
```

Comment:
I changed it to pad_mode because this is what we call it now in separable_filter. But I am also happy to change it back to padding_mode. Then we should also change it in separable_filter...

Reply:
up to you